homeassistant/components/airzone_cloud/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for the Airzone Cloud binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aioairzone_cloud.const import AZD_NAME, AZD_PROBLEMS, AZD_WARNINGS, AZD_ZONES
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+
+@dataclass
+class AirzoneBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes Airzone Cloud binary sensor entities."""
+
+    attributes: dict[str, str] | None = None
+
+
+ZONE_BINARY_SENSOR_TYPES: Final[tuple[AirzoneBinarySensorEntityDescription, ...]] = (
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "warnings": AZD_WARNINGS,
+        },
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        key=AZD_PROBLEMS,
+        name="Problem",
```

Comment:
Take the opportunity to implement translations via `translation_key` instead of setting just the name

Reply:
We don't have to use translation keys for binary sensors; instead, we can rely on the device class to take care of the transaction. Just remove the name should be sufficient.

Reply:
Ok, I'll remove `name` then.

Reply:
@gjohansson-ST @frenck so I tried removing the name, and now `binary_sensor.salon_undefinedtype_singleton` is created instead of `binary_sensor.salon_problem`.
I tried setting `has_entity_name` but it doesn't help either...
So what I should I do in this case?

Reply:
@gjohansson-ST @frenck I tried dropping `name` from the entity_description, but that triggers this warning and doesn't add "_problem" suffix to the entity name from the device_class:
https://github.com/home-assistant/core/blob/86792fcc2f7241f002d629b90fb6bed4fb0177b5/homeassistant/helpers/entity.py#L337-L345

So I added `has_entity_name=True` and `name="Problem"` in the entity description and dropped _attr_name from the BinarySensorEntity.
I'm not sure if this is the proper way to go but I couldn't find any example in homeassistant/components which uses BinarySensorEntityDescription with `has_entity_name=True` and no `name`...

Reply:
@emontnemery since you implemented https://github.com/home-assistant/core/pull/92940 you might be able to shed some light here...
So if I remove `name="Problem"` from the `BinarySensorEntityDescription`, the created entity will be `binary_sensor.salon` instead of `binary_sensor.salon`.
I want to rely on `device_class` in order to take care of the translation but still have the device class appended to the entity name.
Is this possible?
Thanks!

Reply:
You write `binary_sensor.salon` twice so not entirely sure what you mean but my understanding is that if you want it to be `binary_sensor.salon_problem` you need to set `name=None`

Reply:
> You write `binary_sensor.salon` twice so not entirely sure what you mean but my understanding is that if you want it to be `binary_sensor.salon_problem` you need to set `name=None`

You're right (message edited, sorry for that).
I tried that but it still creates `binary_sensor.salon` instead of `binary_sensor.salon_problem`...

Reply:
@gjohansson-ST `name=None` added in db68d059c48f4944158b471f3037b15a105ef4ea
And it fails because `binary_sensor.salon` is created instead of `binary_sensor.salon_problem`:
https://github.com/home-assistant/core/actions/runs/5350411862/jobs/9703046860

Reply:
Fixed by:
https://github.com/home-assistant/core/pull/93583/commits/73ed255ffff436e8e3543c3c2cd1583a60d95cab
And a new merge from upstream (it fails without it):
https://github.com/home-assistant/core/pull/93583/commits/ddb886fdf38c166decd1c062dbd64be0325a5fa9